packages/peregrine/lib/talons/MagentoRoute/useMagentoRoute.js

```diff
@@ -55,49 +26,80 @@ export const useMagentoRoute = props => {
         };
     }, []);
 
-    // If the entry for this pathname is a redirect, perform the redirect.
-    useEffect(() => {
-        if (routeData && routeData.isRedirect) {
-            history.replace(routeData.relativeUrl);
-        }
-    }, [componentMap, history, pathname, routeData]);
+    const setComponent = useCallback(
+        (key, value) => {
+            if (isMountedRef.current) {
```

Comment:
I'm wondering if we should queue these if not mounted yet.

Once we mount, we can iterate over the queue to ensure we process all of these.

I don't have a particular bug in mind that this would solve but this smells like potential for a "silent error" scenario.

Reply:
Actually, this brings up a very good point. Once we visit a client-only page that isn't rendered with `MagentoRoute`, such as `/cart` or `/checkout`, `MagentoRoute` unmounts and we lose its local state. That's why this safeguard is here, but that's also a sign that we should be saving this map elsewhere—somewhere that won't unmount.

We've kept these values in local state because they're functions, so they can't go in local storage. But we could lift them up, at which point we'd no longer lose the map when we navigate (nor need this safeguard). Maybe an Apollo reactive variable?